src/wp-includes/post.php

```diff
@@ -7893,6 +7893,27 @@ function _prime_post_caches( $ids, $update_term_cache = true, $update_meta_cache
 
 		update_post_caches( $fresh_posts, 'any', $update_term_cache, $update_meta_cache );
 	}
+
+	if ( ! $update_meta_cache && ! $update_term_cache ) {
+		return;
+	}
+
+	$ids_may_have_uncached_data = array_diff( $ids, $non_cached_ids );
```

Comment:
What if _prime_post_caches is called twice?

First with `_prime_post_caches( $ids, false, false )`; Then with `_prime_post_caches( $ids, true, true )`; Posts would be in cache but would not cache anything. 

This is a long standing issue, but I would love to fix it here.

Reply:
Unless I'm missing something, that's actually the case we're trying to fix here. `_prime_post_caches( $ids, false, false )` is more or less the same as calling `get_post( $id )`.